httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/DefaultAsyncClientConnectionOperator.java

```diff
@@ -104,7 +107,15 @@ public Future<ManagedAsyncClientConnection> connect(
         Args.notNull(endpointHost, "Host");
         final ComplexFuture<ManagedAsyncClientConnection> future = new ComplexFuture<>(callback);
         final HttpHost remoteEndpoint = RoutingSupport.normalize(endpointHost, schemePortResolver);
-        final InetAddress remoteAddress = endpointHost.getAddress();
+        final Path unixDomainSocket = ((HttpClientContext) context).getRequestConfigOrDefault().getUnixDomainSocket();
```

Comment:
@rschmitt This is quite ugly. Can we introduce a new default `#connect` method here the same way as with `HttpClientConnectionOperator`?